.github/workflows/publish.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow is triggered two ways:
+#
+# 1. When a tag is created, the workflow will upload the package to
+#    test.pypi.org.
+# 2. When a release is made, the workflow will upload the package to pypi.org.
+#
+# It is done this way until PyPI has draft reviews, to allow for a two-stage
+# upload with a chance for manual intervention before the final publication.
+name: Upload package
+
+on:
+  release:
+    types: [created]
+  push:
+    tags:
+      - '*'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: "__token__"
+        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}
+      run: |
+        if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+          export TWINE_REPOSITORY_URL="https://test.pypi.org/legacy/"
+          export TWINE_PASSWORD="${{ secrets.TEST_PYPI_UPLOAD_TOKEN }}"
+        elif [[ "$GITHUB_EVENT_NAME" == "release" ]]; then
+          export TWINE_REPOSITORY_URL="https://upload.pypi.org/legacy/"
+          export TWINE_PASSWORD="${{ secrets.PYPI_UPLOAD_TOKEN }}"
+        else
+          echo "Unknown event name: ${GITHUB_EVENT_NAME}"
+          exit 1
+        fi
```

Comment:
This is overengineering. Have you seen my guide @ https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/ ?

I normally prefer separate steps with if-clauses for readability + it looks way better in the log: steps are separately timed and the output is separated.

Reply:
That guide seems to be doing something different here. I only ever want *exactly* one of these to run (and in fact I'll get rid of all of the TestPyPI stuff as soon as draft releases are available in Warehouse).

I don't think there's a way to do if/elif/else with steps like that, and I'm not sure how beneficial it would be anyway.

Reply:
@pganssle yes, it's slightly different but you can still use if-clauses on the step level which is what I'm advocating for.

Reply:
Alright, let me send you a suggested change so that you'll see what I mean. Also, I don't like too much bash here.